engine/src/main/java/com/datatorrent/stram/api/StreamingContainerUmbilicalProtocol.java

```diff
@@ -341,6 +341,24 @@ public String toString()
     }
   }
 
+  enum ShutdownType
+  {
+    /**
+     * USER_SHUTDOWN, when this value is set all input operators in the
+     * container terminates and send END_STREAM tuple to downstream
+     * operators. This type of shutdown make sure that data emitted by upstream
+     * operator before shutdown are processed by downstreams operator before
+     * they terminate.
+     */
+    USER_SHUTDOWN,
```

Comment:
I would prefer for these options to reflect the "what" vs. where they are currently used from. How about standard terminology, such as WAIT_TERMINATE for this and IMMEDIATE, ABORT or CANCEL for below?